### ISSUE TITLE
Fast yet sensitive way to all-vs-all UR50 search

### ISSUE THREAD

**Issue by Furkan9015** [bbuchfink/diamond#911]
Hi! I am trying to run the following command:

`diamond blastp -q uniref50.fasta -d diamond/uniref50 -f 6 –header -k 200000
–max-hsps 1 -e 0.001 -p 96 -o output.tab`

I am using a single n2-highmem-32 on GCP with 32 CPUs and 256GB RAM. However, it gets stuck in "Computing alignments..." step for hours:

`Processing query block 1, reference block 1/9, shape 2/2, index chunk 4/4.
Building reference seed array...  [2.489s]
Building query seed array...  [2.385s]
Computing hash join...  [0.985s]
Masking low complexity seeds...  [1.292s]
Searching alignments...  [73.662s]
Deallocating memory...  [0s]
Deallocating buffers...  [0.64s]
Clearing query masking...  [0.746s]
Opening temporary output file...  [0s]
Computing alignments... 
`

Any ideas on how to optimize this towards being faster but still decently sensitive?

I am also wondering about the estimated size as I also have a storage concern.

**Comment by Furkan9015** [comment]
Here is how the database looks like: 

`Database: /tmp/poet2_preprocessing/diamond/uniref50 (type: Diamond database, sequences: 58875981, letters: 16875738790)`

I am trying to reproduce PoET-1/2 (https://arxiv.org/abs/2508.04724) training data following

> Models were trained on 62 million sets of homologous sequences. Each set corresponds to a sequence in UniRef50 Version 2304, and contains all its homologs in UniRef50 found using Diamond [34]. To avoid overfitting on promiscuous sequences which may belong to a large number of sets, we sample each set with weight inversely proportional to the size of the set ("inverse count" sequence weighting).

@bbuchfink Also, thanks for the great work.

**Comment by bbuchfink** [comment]
That means the minimal output fields should be ok, as far as I understand. I'd probably recommend a higher sensitivity than default for this application.
